### DISCUSSION TITLE
Modelling the whole EMMO with EMMOntoPy ?

### DISCUSSION THREAD

**Discussion by markdoerr** [emmo-repo/EMMO-python#590] [Ideas]
Dear Developers,

I raised this idea also the the central EMMO repository, but since there is no Discussion Forum, I also want to address this idea here:
What do think about modelling the whole EMMO with EMMOntoPy ?

Why ?
 - EMMOntoPy can be used to modularise the whole EMMO structure when it grows bigger
 -  EMMOntoPy is very expressive and should make Protege superfluous
 - having one, python based toolchain for everything (also for extending EMMO with domain ontologies), with the own IDE of choice.
 - this will provide an extremely good example on how to extend EMMO - illustrating best practices - and demonstrates how to build a   modular ontology 
 - using the CI/CD pipeline / github actions to generate ttl / owl / documentation / reasoning /consistency checks / validation / graphs  automatically

What do you think about this idea ?

**Comment by jesper-friis** [comment]
It is a very good question that should be discussed. EMMO has many authors. Currently, I think that Protege is more accessible for most of the authors, so I wont expect a short-term switch from Protege to EMMOntoPy.

However, EMMOntoPy is already used to generate some of the modules, like periodictable and the modules related to the updated metrology (still in development, but will soon be merged into branch 1.0.0-beta4).

**Comment by markdoerr** [comment]
I fully agree, @jesper-friis, this will be a long-term project. I would be happy, if you can address this in your next EMMO meeting :)
